spec/system/target_spec.rb

```diff
@@ -127,20 +127,99 @@ module Cosmos
+          tgt_filename = File.join(tgt_lib_dir, 'tgt_file.rb')
+          File.open(tgt_filename, 'w') do |file|
+            file.puts "class TgtFile"
+            file.puts "  def self.location; 'tgt'; end"
+            file.puts "end"
+          end
+          lib_filename = File.join(Cosmos::USERPATH, 'lib', 'tgt_file.rb')
+          File.open(lib_filename, 'w') do |file|
+            file.puts "class TgtFile"
+            file.puts "  def self.location; 'lib'; end"
+            file.puts "end"
+          end
+
+          tf = Tempfile.new('unittest')
+          tf.puts("REQUIRE tgt_file.rb")
+          tf.close
+          Target.new("TEST").process_file(tf.path)
+          expect { TgtFile.location }.to raise_error(NameError)
+          expect { Cosmos::TgtFile.location }.to raise_error(NameError)
+          expect(Cosmos::TEST::TgtFile.location).to eql 'tgt'
+          File.delete lib_filename
+          FileUtils.rm_rf File.join(tgt_lib_dir, '..')
+          tf.unlink
+        end
+
+        it "namespaces files in target lib" do
+          lib_filename = File.join(Cosmos::USERPATH, 'lib', 'lib_file.rb')
+          File.open(lib_filename, 'w') do |file|
+            file.puts "class LibFile"
+            file.puts "  def self.location; 'lib'; end"
+            file.puts "end"
+          end
+
+          tgt_lib_dir = File.join(Cosmos::USERPATH, 'config', 'targets', 'TEST', 'lib')
+          FileUtils.mkdir_p(tgt_lib_dir)
+          tgt_filename = File.join(tgt_lib_dir, 'tgt_file.rb')
+          File.open(tgt_filename, 'w') do |file|
+            file.puts "require 'lib_file.rb'" # Verify we can require & use files in lib
+            file.puts "class TgtFile"
+            file.puts "  def self.location; LibFile.location + 'test'; end"
+            file.puts "end"
+          end
+          tgt2_lib_dir = File.join(Cosmos::USERPATH, 'config', 'targets', 'TEST2', 'lib')
+          FileUtils.mkdir_p(tgt2_lib_dir)
+          tgt2_filename = File.join(tgt2_lib_dir, 'tgt_file.rb')
+          File.open(tgt2_filename, 'w') do |file|
+            file.puts "class TgtFile"
+            file.puts "  def self.location; 'test2'; end"
+            file.puts "end"
+          end
+          tgt2_filename = File.join(tgt2_lib_dir, 'tgt2_file.rb')
+          File.open(tgt2_filename, 'w') do |file|
+            file.puts "class TgtFile" # Override the original
+            file.puts "  def self.location; 'test22'; end"
+            file.puts "end"
+          end
+
+          tf1 = Tempfile.new('unittest')
+          tf1.puts("REQUIRE tgt_file.rb")
+          tf1.close
+          tf2 = Tempfile.new('unittest')
+          tf2.puts("REQUIRE tgt_file.rb")
+          tf2.puts("REQUIRE tgt2_file.rb")
+          tf2.close
+          Target.new("TEST").process_file(tf1.path)
+          expect(TEST::TgtFile.location).to eql 'libtest'
+          Target.new("TEST2").process_file(tf2.path)
+          expect(TEST2::TgtFile.location).to eql 'test22'
```

Comment:
Note the targets are created in the namespace as all caps just like how we do it in COSMOS. That's not idiomatic Ruby of course but I think that's good in that target names won't collide with any of our core COSMOS classes.